sbi/inference/posteriors/direct_posterior.py

```diff
@@ -416,16 +417,15 @@ def np_potential(self, theta: np.ndarray) -> ScalarFloat:
         Returns:
             Posterior log probability $\log(p(\theta|x))$.
         """
-
         theta = torch.as_tensor(theta, dtype=torch.float32)
-
-        is_within_prior = torch.isfinite(self.prior.log_prob(theta))
-        if is_within_prior:
-            target_log_prob = self.posterior_nn.log_prob(
-                inputs=theta.reshape(1, -1), context=self.x.reshape(1, -1),
-            )
-        else:
-            target_log_prob = -float("Inf")
+        theta = ensure_theta_batched(theta)
+        num_batch = theta.shape[0]
+        x = ensure_x_batched(self.x).repeat(num_batch, 1)
+
+        with torch.set_grad_enabled(False):
```

Comment:
Why is this needed? Should be taken care of here: https://github.com/mackelab/sbi/blob/fc778f97b79cd13bc198bf14663e1725571c0ec9/sbi/inference/posteriors/base_posterior.py#L315

Reply:
See above